### PR TITLE
add Sonar lint to backend

### PR DESCRIPTION
**Pull request by TSC21**
I am still not sure on how this gets added into SonarCloud. @thirtytwobits maybe you can give an insight?

### CODE REVIEW

**Comment by pavel-kirienko** [comment]
@TSC21 can you see this at the top-right at sonarcloud.io?

image

**Comment by TSC21** [comment]
> @TSC21 can you see this at the top-right at sonarcloud.io?
> 
> image
> 
> And then this?
> 
> image

Yes and as I explained in Slack, I cannot select the UAVCAN organisation. And I think it's because I am not an admin. So rather you or @thirtytwobits need to add it.

File: .sonarcloud.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# For Auto-scan/Pull-Request Scan.
```
Comment (thirtytwobits) [inline .sonarcloud.properties:1]: I'm not sure we want to do automatic scans. I set this up in Nunavut to try to get PR scanning to work but it didn't. See https://sonarcloud.io/documentation/analysis/pull-request/ for how to enable PR scanning correctly.

Comment (TSC21) [reply .sonarcloud.properties:1]: OK

Comment (thirtytwobits) [reply .sonarcloud.properties:1]: As discussed in slack, let's just yank this file for this PR.

**Pull request merged**